### DISCUSSION TITLE
I have issue use LucidRow for soft delete on adonis 6

### DISCUSSION THREAD

**Discussion by Alnsentry** [adonisjs/core#4541] [Help]
the error is Property 'deletedAt' does not exist on type 'LucidRow'. Did you mean 'delete'? ts(2551)

```
import { BaseModel } from '@adonisjs/lucid/orm'
import { LucidRow, ModelQueryBuilderContract } from '@adonisjs/lucid/types/model'
import { DateTime } from 'luxon'

export const softDeleteQuery = (query: ModelQueryBuilderContract<typeof BaseModel>) => {
  query.whereNull(`${query.model.table}.deletedAt`)
}

export const softDelete = async (row: LucidRow, column: string = 'deletedAt') => {
  row['deletedAt'] = DateTime.now()
  await row.save()
}
```

**Comment by sriramk3** [comment]
First: If you are using BaseModel out of the box, whenever you try to query, adonis uses snake case. To over come this, first create your own Strategy to use Camel case which extends base model instead of snake case and use that in place of base model.

Second: For soft delete you can use the npm package: https://www.npmjs.com/package/adonis-lucid-soft-deletes . Out of the box only delete is supported as far as Ik. And also ensure the table you migrated has the deleted at column.
